lib/Doctrine/Hydrator/RecordDriver.php

```diff
@@ -118,7 +118,7 @@ public function flush()
             $coll->takeSnapshot();
         }
         $this->_initializedRelations = null;
-        $this->_collections = null;
+        $this->_collections = array();
```

Comment:
This is not really related to 8.1. Flushing twice caused a warning due to the `$_collection` being set to `null`.